src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -340,33 +340,29 @@ void parse_node_info_linux(const std::vector<std::string> node_info_table,
         int core_1 = 0;
         int core_2 = 0;
         std::string::size_type pos = 0;
-        std::string::size_type endpos = 0;
+        std::string::size_type endpos_1 = 0;
+        std::string::size_type endpos_2 = 0;
         std::string sub_str = "";
 
-        if (((endpos = one_info.find('-', pos)) == std::string::npos) &&
-            ((endpos = one_info.find(',', pos)) != std::string::npos)) {
-            while (endpos != std::string::npos) {
+        while (pos != std::string::npos) {
+            endpos_1 = one_info.find(',', pos);
+            endpos_2 = one_info.find('-', pos);
+            if ((endpos_1 < endpos_2) || (endpos_1 == std::string::npos && endpos_2 == std::string::npos)) {
                 sub_str = one_info.substr(pos);
                 core_1 = std::stoi(sub_str);
                 nodes_table.push_back({core_1, core_1, node_index});
-                endpos = one_info.find(',', pos);
-                pos = endpos + 1;
+                pos = endpos_1 + 1;
```

Comment:
If endpos_1 == std::string::npos, is pos=0? Is it the result we wanted?

Reply:
updated